redash/query_runner/mongodb.py

```diff
@@ -113,13 +113,14 @@ def parse_results(results):
 
         parsed_row = _parse_dict(row)
         for column_name, value in parsed_row.items():
-            columns.append(
-                {
-                    "name": column_name,
-                    "friendly_name": column_name,
-                    "type": TYPES_MAP.get(type(value), TYPE_STRING),
-                }
-            )
+            if _get_column_by_name(columns, column_name) is None:
```

Comment:
To make this more efficient, can't we just get the first row, and then generate the columns array from that outside for this for loop? 

In this case every cell in the table requires this comparison which is in-efficient and wether we check row1 or rowN the columns will be same.

Reply:
@clearnote01 I understand your comment. I also think that if it is normal RDB data, I should do that.

I thought that MongoDB might have a different column for each row, so I implemented it like this. Does such a case not exist?

Reply:
@masayuki038 You are right, I don't think it's a problem when projection is specifically configured via $project, but there could be other cases that would break it. 

Maybe it's safer to keep it as it is, and have a note that this could be optimized in the future if someone can rule out the edge cases.